internal/espressoreader/espresso_reader.go

```diff
@@ -54,10 +58,16 @@ func (e *EspressoReader) Run(ctx context.Context, ready chan<- struct{}) error {
 			return ctx.Err()
 		default:
 			// fetch latest espresso block height
-			latestBlockHeight, err := e.client.FetchLatestBlockHeight(ctx)
+			latestBlockHeight, err := retry.CallFunctionWithRetryPolicy(
+				e.client.FetchLatestBlockHeight,
+				ctx,
+				e.maxRetries,
+				time.Duration(e.maxDelay),
+				"EspressoReader::FetchLatestBlockHeight",
+			)
 			if err != nil {
 				slog.Error("failed fetching latest espresso block height", "error", err)
-				continue
+				os.Exit(1)
```

Comment:
Why doesn't it return with an error? Isn't it too strong to exit the whole thing since this is not `main.go`?